engine/proofview.ml

```diff
@@ -451,6 +451,25 @@ let iter_goal i =
   Solution.get >>= fun evd ->
   Comb.set CList.(undefined evd (flatten (rev subgoals)))
 
+(** List iter but allocates a list of results *)
+let map_goal i =
```

Comment:
Ok.

It would be better to factor that with `iter_goal` but if I remember correctly `iter_goal` is a rather sensitive tactic. So let's keep that for later.